testlapack/matgen.go

```diff
@@ -0,0 +1,118 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Dlatm1 computes the entries of d as specified by mode, cond and rsign.
+//
+// mode describes how d will be computed:
+//  |mode| == 1: d[0] = 1 and d[1:n] = 1/cond
+//  |mode| == 2: d[:n-1] = 1/cond and d[n-1] = 1
+//  |mode| == 3: d[i] = cond^{-i/(n-1)}, i=0,...,n-1
+//  |mode| == 4: d[i] = 1 - i*(1-1/cond)/(n-1)
+//  |mode| == 5: d[i] = random number in the range (1/cond, 1) such that
+//                    their logarithms are uniformly distributed
+//  |mode| == 6: d[i] = random number from the distribution given by dist
+// If mode is negative, the order of the elements of d will be reversed.
+// For other values of mode Dlatm1 will panic.
+//
+// if rsign is true and mode is not ±6, each entry of d will be multiplied by 1
+// or -1 with probability 0.5
+//
+// dist specifies the type of distribution to be used when mode == ±6:
+//  dist == 1: Uniform(0,1)
+//  dist == 2: Uniform(-1,1)
+//  dist == 3: Normal(0,1)
+// For other values of dist Dlatm1 will panic.
+//
+// rnd is used as a source of random numbers.
+func Dlatm1(mode int, cond float64, rsign bool, dist int, rnd *rand.Rand, d []float64) {
```

Comment:
If we're not following the LAPACK API tightly, move `d` to the front and rename to `dst`?